src/machines/child_workflow_state_machine.rs

```diff
@@ -111,3 +340,487 @@ pub(super) struct Terminated {}
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 1);
+        assert_eq!(
+            commands[0].command_type,
+            CommandType::StartChildWorkflowExecution as i32
+        );
+
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 0);
+
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 1);
+        assert_eq!(
+            commands[0].command_type,
+            CommandType::CompleteWorkflowExecution as i32
+        );
+        wfm.shutdown().await.unwrap();
+    }
+
+    #[rstest(wfm, case::start_failure(child_workflow_start_fail_hist()))]
+    #[tokio::test]
+    async fn single_child_workflow_start_fail(mut wfm: ManagedWFFunc) {
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 1);
+        assert_eq!(
+            commands[0].command_type,
+            CommandType::StartChildWorkflowExecution as i32
+        );
+
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 1);
+        assert_eq!(
+            commands[0].command_type,
+            CommandType::CompleteWorkflowExecution as i32
+        );
+        wfm.shutdown().await.unwrap();
+    }
+
+    async fn cancel_before_send_wf(mut ctx: WfContext) -> WorkflowResult<()> {
+        let workflow_id = "child-id-1";
+        let start = ctx.start_child_workflow(StartChildWorkflowExecution {
+            workflow_id: workflow_id.to_string(),
+            workflow_type: "child".to_string(),
+            input: vec![],
+            ..Default::default()
+        });
+        ctx.cancel_child_workflow(workflow_id);
+        match start.await {
+            StartStatus::Cancelled(_) => Ok(().into()),
+            _ => Err(anyhow!("Unexpected start status")),
+        }
+    }
+
+    #[fixture]
+    fn child_workflow_cancel_before_sent() -> ManagedWFFunc {
+        let func = WorkflowFunction::new(cancel_before_send_wf);
+        let mut t = TestHistoryBuilder::default();
+        t.add_by_type(EventType::WorkflowExecutionStarted);
+        t.add_full_wf_task();
+        t.add_workflow_task_scheduled_and_started();
+        assert_eq!(2, t.get_full_history_info().unwrap().wf_task_count());
+        ManagedWFFunc::new(t, func, vec![])
+    }
+
+    #[rstest(wfm, case::default(child_workflow_cancel_before_sent()))]
+    #[tokio::test]
+    async fn single_child_workflow_cancel_before_sent(mut wfm: ManagedWFFunc) {
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 0);
```

Comment:
Another curious seemingly unnecessary activation

Reply:
See my comment above.

Reply:
It's totally valid for a WF not to produce any commands during an activation, one obvious case is for waiting for multiple things to resolve.

Reply:
Right, yes, that makes sense. OK That sounds good to me